receiver/sapnetweaverreceiver/metrics.go

```diff
@@ -40,519 +40,594 @@ var (
 	if err != nil {
 		errs.AddPartial(1, err)
 		return
 	}
 }
 
-func (s *sapNetweaverScraper) recordSapnetweaverIcmAvailabilityDataPoint(now pcommon.Timestamp, alertTreeResponse map[string]string, errs *scrapererror.ScrapeErrors) {
-	metricName := "ICM"
+func (s *sapNetweaverScraper) recordSapnetweaverAbapUpdateErrorCountDataPoint(now pcommon.Timestamp, alertTreeResponse map[string]string, errs *scrapererror.ScrapeErrors) {
+	metricName := "AbapErrorInUpdate"
 	val, ok := alertTreeResponse[metricName]
 	if !ok {
 		errs.AddPartial(1, fmt.Errorf(collectMetricError, metricName, errValueNotFound))
 		return
 	}
-
-	switch models.StateColor(val) {
+	var gray, green, yellow, red int64
+	switch models.StateColor(models.StateColor(val)) {
 	case models.StateColorGray:
-		s.mb.RecordSapnetweaverIcmAvailabilityDataPoint(now, 1, metadata.AttributeControlStateGrey)
-		s.mb.RecordSapnetweaverIcmAvailabilityDataPoint(now, 0, metadata.AttributeControlStateGreen)
-		s.mb.RecordSapnetweaverIcmAvailabilityDataPoint(now, 0, metadata.AttributeControlStateYellow)
-		s.mb.RecordSapnetweaverIcmAvailabilityDataPoint(now, 0, metadata.AttributeControlStateRed)
+		gray = 1
 	case models.StateColorGreen:
-		s.mb.RecordSapnetweaverIcmAvailabilityDataPoint(now, 0, metadata.AttributeControlStateGrey)
-		s.mb.RecordSapnetweaverIcmAvailabilityDataPoint(now, 1, metadata.AttributeControlStateGreen)
-		s.mb.RecordSapnetweaverIcmAvailabilityDataPoint(now, 0, metadata.AttributeControlStateYellow)
-		s.mb.RecordSapnetweaverIcmAvailabilityDataPoint(now, 0, metadata.AttributeControlStateRed)
+		green = 1
 	case models.StateColorYellow:
-		s.mb.RecordSapnetweaverIcmAvailabilityDataPoint(now, 0, metadata.AttributeControlStateGrey)
-		s.mb.RecordSapnetweaverIcmAvailabilityDataPoint(now, 0, metadata.AttributeControlStateGreen)
-		s.mb.RecordSapnetweaverIcmAvailabilityDataPoint(now, 1, metadata.AttributeControlStateYellow)
-		s.mb.RecordSapnetweaverIcmAvailabilityDataPoint(now, 0, metadata.AttributeControlStateRed)
+		yellow = 1
 	case models.StateColorRed:
-		s.mb.RecordSapnetweaverIcmAvailabilityDataPoint(now, 0, metadata.AttributeControlStateGrey)
-		s.mb.RecordSapnetweaverIcmAvailabilityDataPoint(now, 0, metadata.AttributeControlStateGreen)
-		s.mb.RecordSapnetweaverIcmAvailabilityDataPoint(now, 0, metadata.AttributeControlStateYellow)
-		s.mb.RecordSapnetweaverIcmAvailabilityDataPoint(now, 1, metadata.AttributeControlStateRed)
-	default:
-		errs.AddPartial(1, fmt.Errorf(collectMetricError, metricName, errInvalidStateColor))
+		red = 1
 	}
+	s.mb.RecordSapnetweaverAbapUpdateErrorCountDataPoint(now, gray, metadata.AttributeControlStateGray)
+	s.mb.RecordSapnetweaverAbapUpdateErrorCountDataPoint(now, green, metadata.AttributeControlStateGreen)
+	s.mb.RecordSapnetweaverAbapUpdateErrorCountDataPoint(now, yellow, metadata.AttributeControlStateYellow)
+	s.mb.RecordSapnetweaverAbapUpdateErrorCountDataPoint(now, red, metadata.AttributeControlStateRed)
 }
```

Comment:
Good catch, I updated the metric name and description. This metric is weird and produces text of an error message for a value. At first I thought we could count of the length of the error messages, but there is a size limit so that won't work. So only the status can be taken. I have updated the metric description and name to `sapnetweaver.abap.update.status` with a description of `The status of the ABAP updated process.  For context, this is important because an error in AbapErrorInUpdate message specifically indicates that an error occurred during an update operation and that a transaction was not committed.